jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -156,76 +154,67 @@ public boolean isSpatRecord() {
-				} else if (isSpatRecord()) {
-					logger.debug("Publishing a Spat");
-				} else {
-					logger.debug("Publishing a TIM or MAP");
-				}
-
-				if (!(isSpatRecord() && msgMetadata instanceof OdeSpatMetadata
-        			&& !((OdeSpatMetadata) msgMetadata).getIsCertPresent())) {
-					if (checkHeader(msgPayload) == "Ieee1609Dot2Data") {
-						Asn1Encoding msgEncoding = new Asn1Encoding("root", "Ieee1609Dot2Data", EncodingRule.COER);
-						msgMetadata.addEncoding(msgEncoding);
-					}
+				// Determine the message type (MAP or TIM are the current other options)
+				String messageType = determineMessageType(msgPayload);
+				if (messageType == "MAP") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedMAPJson());
+				} else if (messageType == "TIM") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedTIMJson());
 				}
-
-				Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-						EncodingRule.UPER);
-				msgMetadata.addEncoding(unsecuredDataEncoding);
-
-				publisher.publish(xmlUtils.toXml(odeData),
-						publisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
 			}
+
 			serialId.increment();
 		}
 	}
 
-	
 	/**
-		* Checks the header of the OdeMsgPayload and determines the encoding rule to be used in the Asn1DecoderInput XML.
-		* The payload is checked for various message start flags. It will add the encoding rule to the Asn1DecoderInput XML
-		* to tell the ASN1 codec to extract data from the header into the output message.
+		* Determines the message type based off the most likely start flag
 		* 
-		* @param payload The OdeMsgPayload to check the header of.
-		* @return The encoding rule to be used in the Asn1DecoderInput XML.
+		* @param payload The OdeMsgPayload to check the content of.
 		*/
-	public String checkHeader(OdeMsgPayload payload) {
-		JSONObject payloadJson;
-		String header = null;
+	public String determineMessageType(OdeMsgPayload payload) {
+		String messageType = "";
 		try {
-			payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
-			String hexPacket = payloadJson.getString("bytes");
-
-			for (String key : msgStartFlags.keySet()) {
-				String startFlag = msgStartFlags.get(key);
-				int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
-				logger.debug("Start index for " + key + "(" + startFlag + ")" + " is: " + startIndex);
-				if (startIndex <= 20 && startIndex != 0 && startIndex != -1) {
-					logger.debug("Message has supported Ieee1609Dot2Data header, adding encoding rule to Asn1DecoderInput XML");
-					header = "Ieee1609Dot2Data";
-					break;
+			JSONObject payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
+			String hexString = payloadJson.getString("bytes").toLowerCase();
+
+			HashMap<String, Integer> flagIndexes = new HashMap<String, Integer>();
```

Comment:
Does this only support MAP and TIM? If we needs to support more types in future, do we needs to update this piece again? Is it possible to make this generic?

Reply:
It does. Commsignia devices only support writing MAP and TIM messages to a log file so we are specifically implementing this for that use case. We will definitely make this more generic in the future.